code/modules/mob/living/carbon/human/inventory.dm

```diff
@@ -102,7 +102,7 @@
 			if(G.tint)
 				update_tint()
 			if(G.vision_correction)
-				clear_fullscreen("nearsighted")
+				cure_nearsighted()

```

Comment:
Are you sure this fixes it? I believe the issue is that there are two fullscreen overlays for nearsight:
"eye_damage" used here: https://github.com/tgstation/tgstation/blob/8829c1926a0c0de31dc42db612debaa5df3d1d02/code/modules/mob/living/carbon/status_procs.dm#L13
and "nearsighted" used here https://github.com/tgstation/tgstation/blob/8829c1926a0c0de31dc42db612debaa5df3d1d02/code/modules/mob/living/carbon/status_procs.dm#L87
cure_nearsighted() currently only removes the "nearsighted" overlay

Reply:
any other case when curing nearsight, this is the only function that is called.
even in some admin functions.

Reply:
That's because they're meant to permanently remove the disability. This is just temporary overlay removal.